mlir/lib/Dialect/Affine/Analysis/Utils.cpp

```diff
@@ -50,6 +50,27 @@ void LoopNestStateCollector::collect(Operation *opToWalk) {
       loadOpInsts.push_back(op);
     else if (isa<AffineWriteOpInterface>(op))
       storeOpInsts.push_back(op);
```

Comment:
Filtering out ops with alloc and free effects, reason in https://github.com/mcl-csa/llvm-project-public/pull/13#discussion_r1470710250